### PR TITLE
Source S3: updates for compatibility with the concurrent CDK

### PR DESCRIPTION
**Pull request by clnoll**
Add `catalog`, `config`, and `state` to the args sent to the source, for compatibility with the concurrent CDK.

Depends on https://github.com/airbytehq/airbyte/pull/34411.

### CODE REVIEW

File: airbyte-integrations/connectors/source-s3/source_s3/v4/source.py
Comment (girarda) [inline airbyte-integrations/connectors/source-s3/source_s3/v4/source.py:24]: is there value in allowing the user to configure the concurrency level?

**Review by girarda — APPROVED** [review]
looks good! :shipit:

File: airbyte-integrations/connectors/source-s3/source_s3/v4/source.py
Comment (clnoll) [reply airbyte-integrations/connectors/source-s3/source_s3/v4/source.py:24]: This is to allow each source to turn on concurrency intentionally, instead of accidentally deploying it when the next version of the connector is deployed.

**Review by maxi297 — COMMENTED** [review]
CATs will probably fail until https://github.com/airbytehq/airbyte/pull/34605 is available. It is currently blocked by https://github.com/airbytehq/airbyte/pull/34653. I'll release as soon as it is approved

**Pull request merged**
